commonmark-ext-autolink/src/main/java/org/commonmark/ext/autolink/internal/AutolinkPostProcessor.java

```diff
@@ -11,9 +12,40 @@
 
 public class AutolinkPostProcessor implements PostProcessor {
 
-    private LinkExtractor linkExtractor = LinkExtractor.builder()
-            .linkTypes(EnumSet.of(LinkType.URL, LinkType.EMAIL))
-            .build();
+    private final LinkExtractor linkExtractor;
+
+    public AutolinkPostProcessor() {
+        this(EnumSet.of(AutolinkType.URL, AutolinkType.EMAIL));
+    }
+
+    public AutolinkPostProcessor(Set<AutolinkType> linkTypes) {
+        if (linkTypes == null) {
+            throw new NullPointerException("linkTypes must not be null");
+        }
+
+        if (linkTypes.isEmpty()) {
+            throw new IllegalArgumentException("linkTypes must not be empty");
+        }
+
+        EnumSet<LinkType> types = EnumSet.noneOf(LinkType.class);
```

Comment:
We can use `var` nowadays:

```suggestion
        var types = EnumSet.noneOf(LinkType.class);
```